models/annotations_models/train_random_forest_classifier_model.ipynb

```diff
@@ -1062,72 +985,1186 @@
+      "text/plain": [
+       "<IPython.core.display.Javascript object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "data": {
+      "text/html": [
+       "<div id='4adfdd68-dcc1-4f82-84ac-1ccff5ebaa07'></div>"
+      ],
+      "text/plain": [
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'Ground': {'f1-score': 0.9964152263780806,\n",
+      "            'precision': 0.9964399204115688,\n",
+      "            'recall': 0.9963905335685098,\n",
+      "            'support': 2542204},\n",
+      " 'Water': {'f1-score': 0.9980307194135087,\n",
+      "           'precision': 0.9984138968450557,\n",
+      "           'recall': 0.9976478359855142,\n",
+      "           'support': 3814785},\n",
+      " 'Waterplants': {'f1-score': 0.9926857398282446,\n",
+      "                 'precision': 0.9863365540127207,\n",
+      "                 'recall': 0.9991171964195524,\n",
+      "                 'support': 235613}}\n"
+     ]
+    }
+   ],
+   "source": [
+    "final_model = RandomForestClassifier(**optimal_parameters)\n",
+    "final_scaler = StandardScaler()\n",
+    "\n",
+    "train_imbalanced_model(\n",
+    "    X_train=df[selected_features], \n",
+    "    y_train=df[\"label\"], \n",
+    "    model=final_model, \n",
+    "    random_state=1337, \n",
+    "    sampling_type_boundary=sampling_type_boundary ,\n",
+    "    scaler=final_scaler\n",
+    ")\n",
+    "pprint.pprint(get_metrics(y=df[\"label\"], X=df[selected_features], model=final_model, scaler=final_scaler))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "final_artefact = {\n",
+    "    \"model\": final_model,\n",
+    "    \"scaler\": final_scaler\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Saving to ../saved_models/Superview_Nieuwkoopse_plassen_20190302_113613_to_20221012_104900_random_forest_classifier.sav\n"
+     ]
+    }
+   ],
+   "source": [
+    "final_model_filepath = get_model_filepath(location=location, satellite_constellation=satellite_constellation, df=df)\n",
+    "print(f\"Saving to {final_model_filepath}\")\n",
+    "with open(final_model_filepath, \"wb\") as file:\n",
+    "    pickle.dump(final_artefact, file)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Model tests"
```

Comment:
Why is this model tests here? Was it not introduce in a separate test file? If not, why not?

Reply:
we spoke about this, this willl be become a model metrics file

Reply:
which has been made now